src/test/java/com/google/javascript/clutz/goog_scope.d.ts

```diff
@@ -0,0 +1,31 @@
+declare namespace ಠ_ಠ.clutz.foo {
+  class Bar extends Bar_Instance {
+  }
+  class Bar_Instance {
+    private noStructuralTyping_: any;
+  }
+}
+declare namespace ಠ_ಠ.clutz.goog {
+  function require(name: 'foo.Bar'): typeof ಠ_ಠ.clutz.foo.Bar;
+}
+declare module 'goog:foo.Bar' {
+  import alias = ಠ_ಠ.clutz.foo.Bar;
+  export default alias;
```

Comment:
This like should assure that `import Bar from 'goog:foo.Bar'` should work as expected. We need to isolate your failure a bit more.